test/curve_analysis/test_scatter_table.py

```diff
@@ -0,0 +1,270 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test scatter table."""
+
+from test.base import QiskitExperimentsTestCase
+import pandas as pd
+import numpy as np
+
+from qiskit_experiments.curve_analysis.scatter_table import ScatterTable
+
+
+class TestScatterTable(QiskitExperimentsTestCase):
+    """Test cases for curve analysis ScatterTable."""
+
+    def setUp(self):
+        super().setUp()
+
+        source = {
+            "xval": [0.100, 0.100, 0.200, 0.200, 0.100, 0.200, 0.100, 0.200, 0.100, 0.200],
+            "yval": [0.192, 0.784, 0.854, 0.672, 0.567, 0.488, 0.379, 0.671, 0.784, 0.672],
+            "yerr": [0.002, 0.091, 0.090, 0.027, 0.033, 0.038, 0.016, 0.048, 0.091, 0.027],
+            "series_name": [
+                "model1",
+                "model2",
+                "model1",
+                "model2",
+                "model1",
+                "model1",
+                "model1",
+                "model1",
+                "model2",
+                "model2",
+            ],
+            "series_id": [0, 1, 0, 1, 0, 0, 0, 0, 1, 1],
+            "category": [
+                "raw",
+                "raw",
+                "raw",
+                "raw",
+                "raw",
+                "raw",
+                "formatted",
+                "formatted",
+                "formatted",
+                "formatted",
+            ],
+            "shots": [
+                1000,
+                1000,
+                1000,
+                1000,
+                1000,
+                1000,
+                2000,
+                2000,
+                1000,
+                1000,
+            ],
+            "analysis": [
+                "Fit1",
+                "Fit1",
+                "Fit1",
+                "Fit1",
+                "Fit2",
+                "Fit2",
+                "Fit1",
+                "Fit1",
+                "Fit1",
+                "Fit1",
+            ],
+        }
+        self.reference = pd.DataFrame.from_dict(source)
+
+    def test_create_table_from_dataframe(self):
```

Comment:
Since from `from_dataframe` is an alternate constructor, could we do a couple other tests here? The ones I am thinking of are 1. use `self.reference.itertuples()` to iterate over the reference data and call `add_row` on an empty `ScatterTable` and then compare to `ScatterTable.from_dataframe(self.reference)` and 2. for those two scatter tables compare `.__dict__` or at least `.__dict__.keys()` to make sure the alternative constructor does not miss any private attributes from `__init__`.